language/vm/vm_runtime/vm_runtime_types/src/value.rs

```diff
@@ -35,7 +36,7 @@ pub enum Value {
     Address(AccountAddress),
     U64(u64),
     Bool(bool),
-    String(String),
+    UserString(UserString),
```

Comment:
do we need to rename this (and before in Type)? I think the abstract name for that value in the VM should be String not UserString.
Also I am realizing this is changing the value representation in a way I am not clear about.
If we use `Box<String>` we can make the "Value enum" 2 words, with `Box<str>` is 3 words and with `String` is 4 words.
I think we want to leave it open but that may affect the representation of UserString and I am not sure what your thoughts are on that (hope I am making sense). Maybe Box<UserString> would be the answer in that case? or we change the UserString type?

Reply:
Yeah we don't need to rename this -- I can change it back to `String`.

> Also I am realizing this is changing the value representation in a way I am not clear about.

`String` was 3 words while `UserString` is 2.

>  I think we want to leave it open but that may affect the representation of UserString and I am not sure what your thoughts are on that (hope I am making sense). Maybe Box would be the answer in that case? or we change the UserString type?

One of the advantages of having the `UserString` type is that we can change the runtime representation to e.g. `Box<Box<str>>`, or `Box<String>`, by changing just the one spot while keeping the same API.